reporter/base_reporter.go

```diff
@@ -0,0 +1,218 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package reporter // import "go.opentelemetry.io/ebpf-profiler/reporter"
+
+import (
+	"context"
+	"time"
+
+	lru "github.com/elastic/go-freelru"
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/libpf/xsync"
+	"go.opentelemetry.io/ebpf-profiler/reporter/internal/pdata"
+	"go.opentelemetry.io/ebpf-profiler/reporter/internal/samples"
+	semconv "go.opentelemetry.io/otel/semconv/v1.25.0"
+)
+
+// baseReporter encapsulates shared behavior between all the available reporters.
+type baseReporter struct {
+	cfg *Config
+
+	// name is the ScopeProfile's name.
+	name string
+
+	// version is the ScopeProfile's version.
+	version string
+
+	// hostID is the unique identifier of the host.
+	hostID string
+
+	// kernelVersion is the version of the kernel.
+	kernelVersion string
+
+	// hostName is the name of the host.
+	hostName string
+
+	// ipAddress is the IP address of the host.
+	ipAddress string
```

Comment:
While the value of these fields are of interest and exist in the otlp reporter for historical reasons, I think, they should no longer exist in`baseReporter`.
```suggestion
```

In an OTel environment, these four fields should be part of `message ResourceProfile` and components like [resourceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourceprocessor) (I might be wrong on this component, so maybe there is another one that does a better job) should extract and add these attributes.

But for the meantime these fields should still be part of the otlp reporter.

Reply:
Maybe a follow up question: Is `baseReporter` intended to handle pdata processing? So that in a next (?) step an OTel collector uses `baseReporter` and the stand alone version of eBPF profiler remains with OTLP reporter?

Reply:
I believe you're right. I added resources to the collector reported by copying them from the OTLP one. But an additional processor should handle that.